readconf.c

```diff
@@ -1727,25 +1731,57 @@ read_config_file_depth(const char *filename, struct passwd *pw,
 	FILE *f;
 	char line[4096];
 	int linenum;
-	int bad_options = 0;
+	int bad_options = 0;	
 
 	if (depth < 0 || depth > READCONF_MAX_DEPTH)
 		fatal("Too many recursive configuration includes");
 
-	if ((f = fopen(filename, "r")) == NULL)
+	if ((f = fopen(filename, "r")) == NULL) 
 		return 0;
-
-#ifndef WINDOWS /* TODO - implement permission checks for Windows */
 	if (flags & SSHCONF_CHECKPERM) {
+		
+#if WINDOWS
```

Comment:
Add a brief description of why this block has to be Windows specific and what its doing.